### PR TITLE
Document `:inline` option for `rails_command` [ci-skip]

### PR DESCRIPTION
**Pull request by ghousemohamed**
### Summary

The doc for `rails_command` does not provide information about the `:inline` option which can
be passed to the method. This PR adds basic info about this option.

### CODE REVIEW

**Review by jonathanhefner — COMMENTED** [review]
Reiterating my comment from https://github.com/rails/rails/pull/44884#pullrequestreview-941147886:

> My own preference is for the API docs to provide detailed information and limited-scope usage examples, while the guides provide higher-level guidance and cross-cutting usage examples.

Additionally, one of my main motivations for adding the `inline` option was was to [improve performance when testing generators](https://github.com/rails/rails/pull/37516#issue-509668645).  I don't think Rails app templates would benefit to the same extent.  And using `inline: true` [comes with some caveats](https://github.com/rails/rails/pull/37516#issuecomment-544750855).

**Comment by ghousemohamed** [comment]
> Reiterating my comment from [#44884 (review)](https://github.com/rails/rails/pull/44884#pullrequestreview-941147886):
> 
> > My own preference is for the API docs to provide detailed information and limited-scope usage examples, while the guides provide higher-level guidance and cross-cutting usage examples.
> 
> Additionally, one of my main motivations for adding the `inline` option was was to [improve performance when testing generators](https://github.com/rails/rails/pull/37516#issue-509668645). I don't think Rails app templates would benefit to the same extent. And using `inline: true` [comes with some caveats](https://github.com/rails/rails/pull/37516#issuecomment-544750855).

@jonathanhefner Hmm, Misjudged the point here. There's more to the `:inline` option. I was not aware of the full picture with regards to the `:inline` option.

The `:inline` option is not particularly a straightforward option to use, should it be documented nonetheless in the API doc for the `rails_command` method? Right now the `rails_command` method API doc does not say anything about the possible values for the `options` argument. Should it also be taken a step further in documenting all the options the `rails_command` method accepts?

**Comment by jonathanhefner** [comment]
@ghousemohamed  I think documenting the `rails_command` options would be good.  But if we include `inline` in the documentation, we should advise proper caution.

File: guides/source/rails_application_templates.md [hunk 1]
```diff
@@ -201,6 +201,12 @@ You can also run commands that should abort application generation if they fail:
 rails_command "db:migrate", abort_on_failure: true
 ```
 
+You can also run commands with the `inline` option. This will run the command directly, instead of shelling out.
+
+```ruby
```
Comment (bestjane) [inline guides/source/rails_application_templates.md:206]: sadasdad

Comment (bestjane) [reply guides/source/rails_application_templates.md:206]: sadasd

**Review by bestjane — COMMENTED** [review]
LGTM

**Comment by ghousemohamed** [comment]
> @ghousemohamed I think documenting the `rails_command` options would be good. But if we include `inline` in the documentation, we should advise proper caution.

Hmm, ok. I will close this issue, and possibly open a PR separately to add docs for `rails_command`

**Pull request closed without merging**